onebusaway-android/src/main/java/org/onebusaway/android/report/ui/dialog/CustomerServiceDialog.java

```diff
@@ -0,0 +1,168 @@
+
+    private ListView mListView;
+
+    private ObaAgenciesWithCoverageResponse mResponse;
+
+    private Adapter mAdapter;
+
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+
+        getLoaderManager().initLoader(0, null, this);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.report_issue_customer_service, null, false);
+
+        mListView = (ListView) view.findViewById(R.id.list);
+        getDialog().getWindow().requestFeature(Window.FEATURE_NO_TITLE);
+        return view;
+    }
+
+    @Override
+    public Loader<ObaAgenciesWithCoverageResponse> onCreateLoader(int id, Bundle args) {
+        return new AgenciesLoader(getActivity());
+    }
+
+    @Override
+    public void onLoadFinished(Loader<ObaAgenciesWithCoverageResponse> loader, ObaAgenciesWithCoverageResponse data) {
+        // Create our generic adapter
+        mResponse = data;
+        mAdapter = new Adapter(getActivity());
+        mListView.setAdapter(mAdapter);
+        //Process the list check phone & email
+        mAdapter.setData(Arrays.asList(data.getAgencies()));
+    }
+
+    @Override
+    public void onLoaderReset(Loader<ObaAgenciesWithCoverageResponse> loader) {
+        mListView.setAdapter(null);
+        mAdapter = null;
+    }
+
+    private class Adapter extends ArrayAdapter<ObaAgencyWithCoverage> {
+
+        Adapter(Context context) {
+            super(context, R.layout.report_issue_customer_service_item);
+        }
+
+        @Override
+        protected void initView(View view, ObaAgencyWithCoverage coverage) {
+            final ObaAgency agency = mResponse.getAgency(coverage.getId());
+
+            TextView text1 = (TextView) view.findViewById(R.id.ricsi_text);
+            text1.setText(agency.getName());
+
+            ImageButton phoneButton = (ImageButton) view.findViewById(R.id.ricsi_phone);
+            phoneButton.setColorFilter(getActivity().getResources().
+                    getColor(R.color.navdrawer_icon_tint_selected));
+
+            ImageButton webButton = (ImageButton) view.findViewById(R.id.ricsi_web);
+            webButton.setColorFilter(getActivity().getResources().
+                    getColor(R.color.navdrawer_icon_tint_selected));
+
+            ImageButton emailButton = (ImageButton) view.findViewById(R.id.ricsi_email);
+            emailButton.setColorFilter(getActivity().getResources().
+                    getColor(R.color.navdrawer_icon_tint_selected));
+
+            if (TextUtils.isEmpty(agency.getEmail())) {
+                emailButton.setVisibility(View.INVISIBLE);
+            } else {
+                emailButton.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View view) {
```

Comment:
Very important - could you please add analytics to each of the customer service contact methods (email, phone, website)?  We'd definitely want to know how the user is reaching out to the agency, as it will be hard to get metrics about this from the agency itself.